.github/workflows/deploy-pypi.yml

```diff
@@ -0,0 +1,51 @@
+# - Build Python Wheels
+# - Deploy to TestPyPI
+# - Download from TestPyPI
+# - Run tests on downloaded artifacts
+# - Deploy to real PyPI
```

Comment:
[fix] Remove?

Reply:
Yep, no comment at all is really neccesary now as workflow is much less complex.

Reply:
[33fc41d](https://github.com/OpenAssetIO/OpenAssetIO/pull/676/commits/33fc41d0b33b170428cca2d70ea1216ec697fdef)